x/wasm/types/validation.go

```diff
@@ -13,14 +13,17 @@ var (
 
 	// MaxWasmSize is the largest a compiled contract code can be when storing code on chain
 	MaxWasmSize = 800 * 1024 // extension point for chains to customize via compile flag.
+
+	// MaxProposalWasmSize is the largest a gov proposal compiled contract code can be when storing code on chain
+	MaxProposalWasmSize = 4 * 1024 * 1024 // extension point for chains to customize via compile flag.
```

Comment:
Good to have this as a variable, too. I am not fully convinced that 4mb is a good value, given the spam case. 2mb feels a bit more conservative but  it is also a random value

Reply:
gov proposals charge like 3x gas for the proposal size (than just raw tx size), but yes it is written to state

It is easy to get 1.5MB contracts, maybe 2.5MB or 3MB if you want to be a bit conservative but still allow more complexity (like a js interpreter)

Also, we could just charge extra gas for every byte above the base (800kb) level up to this hard cap, but that might be too confusing/complex

Reply:
If chains can override, we just need a reasonable default. I am fine with 3MB. Feel 2MB is a bit tight (I got a demo contract I could only trim to 1.8MB and could easily grow a bit if I added any features)